### PR TITLE
Adding vertical cubes load distance to vanilla video settings

### PR DESCRIPTION
**Pull request by Foghrye4**
Why can’t automatically merge?

### CODE REVIEW

**Comment by Niko-sk2x** [comment]
You somehow managed to have different version of CubicChunks class than the one in this repository.

**Comment by Foghrye4** [comment]
Rebased to head.

File: src/main/java/cubicchunks/CubicChunks.java [hunk 1]
```diff
@@ -155,19 +155,49 @@ public static void addConfigChangeListener(IConfigUpdateListener listener) {
...
+        public static enum Options {
+            maxGeneratedCubesPerTick(1, Integer.MAX_VALUE, 49 * 16, "The number of cubic chunks to generate per tick."),
+            lightingTickBudget(1, Integer.MAX_VALUE, 10,
+                    "The maximum amount of time in milliseconds per tick to spend performing lighting calculations."),
+            verticalCubeLoadDistance(2, 32, 8, "Similar to Minecraft's view distance, only for vertical chunks."),
+            worldHeightLowerBound(AddressTools.MIN_BLOCK_Y, 0, -4096,
+                    "The lower boundary on the world. Blocks will not generate or load below this point."),
+            worldHeightUpperBound(256, AddressTools.MAX_BLOCK_Y, 4096,
+                    "The upper boundary on the world. Blocks will not generate or load above this point."),
+            chunkGCInterval(1, Integer.MAX_VALUE, 20 * 10,
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/CubicChunks.java]: lowercase enums? really? Took me a while to figure out how you are calling a method without being in some method already...

File: src/main/java/cubicchunks/CubicChunks.java [hunk 2]
```diff
@@ -180,50 +210,38 @@ void loadConfig(Configuration configuration) {
...
-                    "The lower boundary on the world. Blocks will not generate or load below this point.");
-            worldHeightUpperBound = configuration.getInt("worldHeightUpperBound", Configuration.CATEGORY_GENERAL,
-                    DEFAULT_MAX_WORLD_HEIGHT, 256, AddressTools.MAX_BLOCK_Y,
-                    "The upper boundary on the world. Blocks will not generate or load above this point.");
-            chunkGCInterval = configuration.getInt("chunkGCInterval", Configuration.CATEGORY_GENERAL,
-                    DEFAULT_CHUNK_GC_INTERVAL, 1, Integer.MAX_VALUE,
-                    "Chunk garbage collector update interval. A more lower it is - a more CPU load it will generate. "
-                    + "A more high it is - a more memory will be used to store cubes between launches.");
+            for(Options configOption:Options.values())
+            {
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/CubicChunks.java]: Formatting...

File: src/main/java/cubicchunks/client/ClientEventHandler.java [hunk 3]
```diff
@@ -47,4 +57,103 @@ public void onWorldClientTickEvent(TickEvent.ClientTickEvent evt) {
...
+            GuiVideoSettings gvs = (GuiVideoSettings)currentGui;
+            GuiOptionsRowList gowl = (GuiOptionsRowList) gvs.optionsRowList;
+            GuiOptionsRowList.Row row = this.createRow(100, CubicChunks.Config.Options.verticalCubeLoadDistance, gvs.width);
+            gowl.options.add(1, row);
+        }
+    }
+    
+    private GuiOptionsRowList.Row createRow(int buttonId, CubicChunks.Config.Options option, int width)
+    {
+        GuiCustomSlider slider = new GuiCustomSlider(buttonId, width / 2 - 155 + 160, 0, option);
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/client/ClientEventHandler.java:73]: More formatting...

File: src/main/resources/assets/cubicchunks/lang/en_US.lang [hunk 4]
```diff
@@ -17,6 +17,13 @@ cubicchunks.gui.customcubic.ores_tab_title=Customize World Settings%nPage 2 of 3
 cubicchunks.gui.customcubic.advanced_tab_title=Customize World Settings%nPage 3 of 3%nAdvanced Settings
 
 #################################
+###        options            ###
+#################################
+
+cubicchunks.gui.verticalCubeLoadDistance=Vertical Distance:
+cubicchunks.gui.cubes=cubes
```
Comment (Niko-sk2x) [inline src/main/resources/assets/cubicchunks/lang/en_US.lang]: These 2 can be combined into `Vertical Distance: %d cubes` and you would just use `I18n.format("cubicchunks.gui." + option.name(), option.getValue())`

**Review by Niko-sk2x — CHANGES_REQUESTED** [review]
Reformat all of the code, and show some screenshot.

**Comment by Foghrye4** [comment]
Aaargh! (test failed)

**Comment by Niko-sk2x** [comment]
I should rewrite these tests, they only cause issues...

**Comment by Foghrye4** [comment]
@Barteks2x test failed because I misplace upper and lower bounds, shame on me. May I skip making screenshots? Trust me, it's working. If you load a world and change value to bigger in singleplayer game, cubes will be loaded.
EDIT: Damn, you was AFC long enough so I had time to make screenshots and upload them.
Imgur
Imgur

**Comment by Niko-sk2x** [comment]
I meant screenshot of the config screen, becase that's what ou actually added.

**Comment by Foghrye4** [comment]
Duh.
Imgur

**Review by Niko-sk2x — APPROVED** [review]

File: src/main/java/cubicchunks/CubicChunks.java [hunk 5]
```diff
@@ -156,18 +156,66 @@ public static void addConfigChangeListener(IConfigUpdateListener listener) {
...
+                }
+            }
+
+            public int getValue() {
+                return value;
+            }
+
+            public String getNicelyFormattedName() {
+                StringBuffer out = new StringBuffer();
+                char char_ = "_".charAt(0);
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/CubicChunks.java]: I'm almost sure there is some library that already does what this method does.

If you want to go this way, why not use the char literal? `'_' == "_".charAt(0)`

Comment (Foghrye4) [reply src/main/java/cubicchunks/CubicChunks.java]: There is such thing? Nice.

**Pull request merged**